homeassistant/components/nextdns/sensor.py

```diff
@@ -0,0 +1,299 @@
+        coordinator_type=ATTR_IP_VERSIONS,
+        entity_category=EntityCategory.DIAGNOSTIC,
+        entity_registry_enabled_default=False,
+        icon="mdi:ip",
+        name="{profile_name} IPv6 Queries",
+        native_unit_of_measurement="queries",
+        state_class=SensorStateClass.MEASUREMENT,
+    ),
+    NextDnsSensorEntityDescription(
+        key="ipv6_queries_ratio",
+        coordinator_type=ATTR_IP_VERSIONS,
+        entity_category=EntityCategory.DIAGNOSTIC,
+        entity_registry_enabled_default=False,
+        icon="mdi:ip",
+        name="{profile_name} IPv6 Queries Ratio",
+        native_unit_of_measurement=PERCENTAGE,
+        state_class=SensorStateClass.MEASUREMENT,
+    ),
+    NextDnsSensorEntityDescription(
+        key="validated_queries",
+        coordinator_type=ATTR_DNSSEC,
+        entity_category=EntityCategory.DIAGNOSTIC,
+        entity_registry_enabled_default=False,
+        icon="mdi:lock-check",
+        name="{profile_name} DNSSEC Validated Queries",
+        native_unit_of_measurement="queries",
+        state_class=SensorStateClass.MEASUREMENT,
+    ),
+    NextDnsSensorEntityDescription(
+        key="not_validated_queries",
+        coordinator_type=ATTR_DNSSEC,
+        entity_category=EntityCategory.DIAGNOSTIC,
+        entity_registry_enabled_default=False,
+        icon="mdi:lock-alert",
+        name="{profile_name} DNSSEC Not Validated Queries",
+        native_unit_of_measurement="queries",
+        state_class=SensorStateClass.MEASUREMENT,
+    ),
+    NextDnsSensorEntityDescription(
+        key="validated_queries_ratio",
+        coordinator_type=ATTR_DNSSEC,
+        entity_category=EntityCategory.DIAGNOSTIC,
+        entity_registry_enabled_default=False,
+        icon="mdi:lock-check",
+        name="{profile_name} DNSSEC Validated Queries Ratio",
+        native_unit_of_measurement=PERCENTAGE,
+        state_class=SensorStateClass.MEASUREMENT,
+    ),
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Add a NextDNS entities from a config_entry."""
+    sensors: list[NextDnsSensor] = []
+    coordinators = hass.data[DOMAIN][entry.entry_id]
+
+    for description in SENSORS:
+        sensors.append(
+            NextDnsSensor(coordinators[description.coordinator_type], description)
+        )
+
+    async_add_entities(sensors)
+
+
+class NextDnsSensor(CoordinatorEntity, SensorEntity):
+    """Define an NextDNS sensor."""
+
+    coordinator: NextDnsUpdateCoordinator
+
+    def __init__(
+        self,
+        coordinator: NextDnsUpdateCoordinator,
+        description: SensorEntityDescription,
+    ) -> None:
+        """Initialize."""
+        super().__init__(coordinator)
+        self._attr_device_info = coordinator.device_info
+        self._attr_unique_id = f"{coordinator.profile_id}_{description.key}"
+        self._attr_name = cast(str, description.name).format(
```

Comment:
I couldn't find where `name` is defined to understand why the cast is needed. Can you point me to where the name is defined in the entity description?

Reply:
`name` is defined here https://github.com/home-assistant/core/blob/dev/homeassistant/helpers/entity.py#L225 and the type of it is `str | None`. `mypy` complains that `Item "None" of "Optional[str]" has no attribute "format"`.